### DISCUSSION TITLE
Figure out some way to test the app without a sound touch speaker

### DISCUSSION THREAD

**Discussion by Suzie97** [syfds/soundy#29] [Ideas]
SoundTouch speakers are not sold officially anymore. And being an elementary OS specific app, it makes the app even more niche.

It will be hard to find testers and users for the app.

Adding some kind of support for generic speakers (bluetooth or analog), might be a nice idea.

**Comment by Suzie97** [comment]
For now, how about implementing a dummy UI for the music player? A separate testing branch can be created for testing the UI and UX of the app without having to pair a SoundTouch speaker.

**Reply by syfds** [reply]
hi @Suzie97 , thanks for starting this discussion!

**support of multiple speaker API's**:
I missed a client for my SoundTouch speaker so I decided to write one. Generic support was not really my intention. Supporting of the new line of bose speakers can be interesting, but like you already said there is no official API for now. 

**dummy ui**: 
Unfortunately I have no statistic about the downloads in elementary AppCenter. So for now it is very hard to say, whether the community is generally interested in such app. But if the amount of app user is growing and there is a interest in such mocking of the API for local tests we can write one. Spontaneous I think about a node app with a http mock server. So every developer or tester is able to connect to a locally running mock server and test the app. 
Did I understand you right that you would create a branch with a mock implementation of the API? Instead http calls specific mock-data will be returned?

**Reply by Suzie97** [reply]
No, I was just talking about the gtk (front end) part. So that I can test and tweak how the app looks and behaves. But your idea is better. 👍️

**Comment by Suzie97** [comment]
@syfds, can you open a reddit post on r/bose, about the app and invite users and tester?? That way, we might get at least a few people to work on the project.

**Reply by syfds** [reply]
@Suzie97 , good idea! I've already created a post on bose SoundTouch API forum:  https://bosedevs.bose.com/discussion/798/unofficial-gtk-client-for-linux-based-operating-systems#latest.

**Reply by Suzie97** [reply]
Also, open up a post in reddit if you can. We might find more people over there.
